akka-docs-new/src/test/scala/tutorial_3/DeviceGroup.scala

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package tutorial_3
+
+import akka.actor.{ Actor, ActorLogging, ActorRef, Cancellable, Props, Stash, Terminated }
+import tutorial_3.DeviceGroup._
+import tutorial_3.DeviceManager.RequestTrackDevice
+import scala.concurrent.duration._
+
+object DeviceGroup {
+
+  def props(groupId: String): Props = Props(new DeviceGroup(groupId))
+
+  case class RequestDeviceList(requestId: Long)
+  case class ReplyDeviceList(requestId: Long, ids: Set[String])
+
+  case class RequestAllTemperatures(requestId: Long)
+  case class RespondAllTemperatures(requestId: Long, temperatures: Map[String, Option[Double]])
+
+  case class CollectionTimeout(requestId: Long)
+}
+
+class DeviceGroup(groupId: String) extends Actor with ActorLogging with Stash {
+  var deviceIdToActor = Map.empty[String, ActorRef]
+  var actorToDeviceId = Map.empty[ActorRef, String]
```

Comment:
I think it is an anti-pattern, these IDs just happen to be Strings right now, but they could be anything, really. I was thinking introducing a wrapper, Id type, but that felt a bit of overkill for the tutorial (it just adds an additional layer to be explained).